from PyQt5 import QtCore, QtGui, QtWidgets
import psycopg2
from cfg import host, user, db_name, password, port


class Ui_Laborant_add_paccient(object):
    def setupUi(self, Laborant_add_paccient):
        Laborant_add_paccient.setObjectName("Medic")
        Laborant_add_paccient.resize(1011, 637)
        icon = QtGui.QIcon("img/1.png")
        Laborant_add_paccient.setWindowIcon(icon)
        Laborant_add_paccient.setObjectName("Laborant_add_paccient")
        Laborant_add_paccient.resize(600, 700)
        self.label_16 = QtWidgets.QLabel(Laborant_add_paccient)
        self.label_16.setEnabled(True)
        self.label_16.setGeometry(QtCore.QRect(70, 90, 231, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(18)
        self.label_16.setFont(font)
        self.label_16.setStyleSheet("")
        self.label_16.setTextFormat(QtCore.Qt.AutoText)
        self.label_16.setAlignment(QtCore.Qt.AlignCenter)
        self.label_16.setObjectName("label_16")
        self.label_17 = QtWidgets.QLabel(Laborant_add_paccient)
        self.label_17.setEnabled(True)
        self.label_17.setGeometry(QtCore.QRect(70, 140, 231, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(18)
        self.label_17.setFont(font)
        self.label_17.setStyleSheet("")
        self.label_17.setTextFormat(QtCore.Qt.AutoText)
        self.label_17.setAlignment(QtCore.Qt.AlignCenter)
        self.label_17.setObjectName("label_17")
        self.lineEdit = QtWidgets.QLineEdit(Laborant_add_paccient)
        self.lineEdit.setGeometry(QtCore.QRect(300, 90, 231, 31))
        self.lineEdit.setObjectName("lineEdit")
        self.lineEdit_3 = QtWidgets.QLineEdit(Laborant_add_paccient)
        self.lineEdit_3.setGeometry(QtCore.QRect(300, 140, 231, 31))
        self.lineEdit_3.setObjectName("lineEdit_3")
        self.label_20 = QtWidgets.QLabel(Laborant_add_paccient)
        self.label_20.setEnabled(True)
        self.label_20.setGeometry(QtCore.QRect(180, 0, 270, 60))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(19)
        self.label_20.setFont(font)
        self.label_20.setStyleSheet("")
        self.label_20.setTextFormat(QtCore.Qt.AutoText)
        self.label_20.setAlignment(QtCore.Qt.AlignCenter)
        self.label_20.setObjectName("label_20")
        self.pushButton_13 = QtWidgets.QPushButton(Laborant_add_paccient)
        self.pushButton_13.setGeometry(QtCore.QRect(220, 630, 181, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(12)
        self.pushButton_13.setFont(font)
        self.pushButton_13.setObjectName("pushButton_13")
        self.label_18 = QtWidgets.QLabel(Laborant_add_paccient)
        self.label_18.setEnabled(True)
        self.label_18.setGeometry(QtCore.QRect(70, 190, 231, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(18)
        self.label_18.setFont(font)
        self.label_18.setStyleSheet("")
        self.label_18.setTextFormat(QtCore.Qt.AutoText)
        self.label_18.setAlignment(QtCore.Qt.AlignCenter)
        self.label_18.setObjectName("label_18")
        self.lineEdit_4 = QtWidgets.QLineEdit(Laborant_add_paccient)
        self.lineEdit_4.setGeometry(QtCore.QRect(300, 190, 231, 31))
        self.lineEdit_4.setObjectName("lineEdit_4")
        self.label_19 = QtWidgets.QLabel(Laborant_add_paccient)
        self.label_19.setEnabled(True)
        self.label_19.setGeometry(QtCore.QRect(70, 240, 231, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(18)
        self.label_19.setFont(font)
        self.label_19.setStyleSheet("")
        self.label_19.setTextFormat(QtCore.Qt.AutoText)
        self.label_19.setAlignment(QtCore.Qt.AlignCenter)
        self.label_19.setObjectName("label_19")
        self.lineEdit_5 = QtWidgets.QLineEdit(Laborant_add_paccient)
        self.lineEdit_5.setGeometry(QtCore.QRect(300, 240, 231, 31))
        self.lineEdit_5.setObjectName("lineEdit_5")
        self.label_21 = QtWidgets.QLabel(Laborant_add_paccient)
        self.label_21.setEnabled(True)
        self.label_21.setGeometry(QtCore.QRect(70, 290, 231, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(18)
        self.label_21.setFont(font)
        self.label_21.setStyleSheet("")
        self.label_21.setTextFormat(QtCore.Qt.AutoText)
        self.label_21.setAlignment(QtCore.Qt.AlignCenter)
        self.label_21.setObjectName("label_21")
        self.lineEdit_6 = QtWidgets.QLineEdit(Laborant_add_paccient)
        self.lineEdit_6.setGeometry(QtCore.QRect(300, 290, 231, 31))
        self.lineEdit_6.setObjectName("lineEdit_6")
        self.label_22 = QtWidgets.QLabel(Laborant_add_paccient)
        self.label_22.setEnabled(True)
        self.label_22.setGeometry(QtCore.QRect(70, 340, 231, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(18)
        self.label_22.setFont(font)
        self.label_22.setStyleSheet("")
        self.label_22.setTextFormat(QtCore.Qt.AutoText)
        self.label_22.setAlignment(QtCore.Qt.AlignCenter)
        self.label_22.setObjectName("label_22")
        self.lineEdit_7 = QtWidgets.QLineEdit(Laborant_add_paccient)
        self.lineEdit_7.setGeometry(QtCore.QRect(300, 340, 231, 31))
        self.lineEdit_7.setObjectName("lineEdit_7")
        self.label_23 = QtWidgets.QLabel(Laborant_add_paccient)
        self.label_23.setEnabled(True)
        self.label_23.setGeometry(QtCore.QRect(70, 390, 231, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(18)
        self.label_23.setFont(font)
        self.label_23.setStyleSheet("")
        self.label_23.setTextFormat(QtCore.Qt.AutoText)
        self.label_23.setAlignment(QtCore.Qt.AlignCenter)
        self.label_23.setObjectName("label_23")
        self.lineEdit_8 = QtWidgets.QLineEdit(Laborant_add_paccient)
        self.lineEdit_8.setGeometry(QtCore.QRect(300, 390, 231, 31))
        self.lineEdit_8.setObjectName("lineEdit_8")
        self.label_24 = QtWidgets.QLabel(Laborant_add_paccient)
        self.label_24.setEnabled(True)
        self.label_24.setGeometry(QtCore.QRect(70, 440, 231, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(13)
        self.label_24.setFont(font)
        self.label_24.setStyleSheet("")
        self.label_24.setTextFormat(QtCore.Qt.AutoText)
        self.label_24.setAlignment(QtCore.Qt.AlignCenter)
        self.label_24.setObjectName("label_24")
        self.lineEdit_9 = QtWidgets.QLineEdit(Laborant_add_paccient)
        self.lineEdit_9.setGeometry(QtCore.QRect(300, 440, 231, 31))
        self.lineEdit_9.setObjectName("lineEdit_9")
        self.label_25 = QtWidgets.QLabel(Laborant_add_paccient)
        self.label_25.setEnabled(True)
        self.label_25.setGeometry(QtCore.QRect(70, 490, 231, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(13)
        self.label_25.setFont(font)
        self.label_25.setStyleSheet("")
        self.label_25.setTextFormat(QtCore.Qt.AutoText)
        self.label_25.setAlignment(QtCore.Qt.AlignCenter)
        self.label_25.setObjectName("label_25")
        self.comboBox = QtWidgets.QComboBox(Laborant_add_paccient)
        self.comboBox.setGeometry(QtCore.QRect(300, 491, 231, 31))
        self.comboBox.setObjectName("comboBox")
        self.label_26 = QtWidgets.QLabel(Laborant_add_paccient)
        self.label_26.setEnabled(True)
        self.label_26.setGeometry(QtCore.QRect(70, 540, 231, 31))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(13)
        self.label_26.setFont(font)
        self.label_26.setStyleSheet("")
        self.label_26.setTextFormat(QtCore.Qt.AutoText)
        self.label_26.setAlignment(QtCore.Qt.AlignCenter)
        self.label_26.setObjectName("label_26")
        self.comboBox_2 = QtWidgets.QComboBox(Laborant_add_paccient)
        self.comboBox_2.setGeometry(QtCore.QRect(300, 540, 231, 31))
        self.comboBox_2.setObjectName("comboBox_2")

        self.pushButton_14 = QtWidgets.QPushButton(Laborant_add_paccient)
        self.pushButton_14.setGeometry(QtCore.QRect(0, 0, 181, 41))
        font = QtGui.QFont()
        font.setFamily("Comic Sans MS")
        font.setPointSize(12)
        self.pushButton_14.setFont(font)
        self.pushButton_14.setObjectName("pushButton_14")


        self.retranslateUi(Laborant_add_paccient)
        QtCore.QMetaObject.connectSlotsByName(Laborant_add_paccient)

    def data_insert(self):
        sename = self.lineEdit.text()
        name = self.lineEdit_3.text()
        birth_day = self.lineEdit_4.text()
        ser_pas = self.lineEdit_5.text()
        nom_pas = self.lineEdit_6.text()
        phone = self.lineEdit_7.text()
        email = self.lineEdit_8.text()
        polis = self.lineEdit_9.text()

        connection = False

        try:
            connection = psycopg2.connect(
                host=host,
                user=user,
                port=port,
                database=db_name,
                password=password
            )

            with connection.cursor() as cursor:
                cursor.execute(
                    f"""INSERT INTO public.patient(
	email, social_sec_number, phone, passport_s, passport_n, birthdate_timestamp, name, lastname)
	VALUES ('{email}', '{polis}', '{phone}', '{ser_pas}', '{nom_pas}', '{birth_day}', '{name}', '{sename}');"""
                )

        except Exception as _ex:
            print("[ИНФО] Ошибка при работе с PostrgeSQL:", _ex)
        finally:
            if connection:
                cursor.close()
                connection.close()


    def retranslateUi(self, Laborant_add_paccient):
        _translate = QtCore.QCoreApplication.translate
        Laborant_add_paccient.setWindowTitle(_translate("Laborant_add_paccient", "Medic"))
        self.label_16.setText(_translate("Laborant_add_paccient", "Фамилия"))
        self.label_17.setText(_translate("Laborant_add_paccient", "Имя"))
        self.label_20.setText(_translate("Laborant_add_paccient", "Добавте пациента"))
        self.pushButton_13.setText(_translate("Laborant_add_paccient", "Добавить пациента"))
        self.pushButton_14.setText(_translate("Laborant_add_paccient", "Вернуться"))
        self.label_18.setText(_translate("Laborant_add_paccient", "Дата рождения"))
        self.label_19.setText(_translate("Laborant_add_paccient", "Серия паспорта"))
        self.label_21.setText(_translate("Laborant_add_paccient", "Номер паспорта"))
        self.label_22.setText(_translate("Laborant_add_paccient", "Телефон"))
        self.label_23.setText(_translate("Laborant_add_paccient", "e-mail"))
        self.label_24.setText(_translate("Laborant_add_paccient", "Номер страхового полиса"))
        self.label_25.setText(_translate("Laborant_add_paccient", "Тип страхового полиса "))
        self.label_26.setText(_translate("Laborant_add_paccient", "Страховая компания"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    Laborant_add_paccient = QtWidgets.QDialog()
    ui = Ui_Laborant_add_paccient()
    ui.setupUi(Laborant_add_paccient)
    Laborant_add_paccient.show()
    sys.exit(app.exec_())
